File: apps/party-time-backend/src/main/java/com/partytime/api/ApiConstants.java
package com.partytime.api;

import lombok.AccessLevel;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
public final class ApiConstants {

    public static final String REGEX_PASSWORD = "^(?=.*[a-z])(?=.*[A-Z])(?=.+[0-9].+)(?=.*[a-zA-Z]+.*[ ,!\\\"§$%&/()=?{}[\\\\].+]+.*[a-zA-Z]+.*).*$";

}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/api/ApiConstants.kt
package com.partytime.api

/**
 * Class containing constant values for the API
 */
class ApiConstants {
    companion object {

        /**
         * Regex string used for password validation
         */
        const val REGEX_PASSWORD: String =
            "^(?=.*[a-z])(?=.*[A-Z])(?=.+[0-9].+)(?=.*[a-zA-Z]+.*[ ,!\\\"§$%&/()=?{}[\\\\].+]+.*[a-zA-Z]+.*).*$"
    }
}
